### PR TITLE
config: generate jobs and config for 4.7

### PR DESCRIPTION
**Pull request by droslean**
Generated with

```sh
#!/bin/bash

set -o nounset
set -o errexit
set -o pipefail

config-brancher --config-dir ci-operator/config --current-release=4.5 --future-release=4.6 --future-release=4.7 --confirm
git add ci-operator/config
git commit --signoff -S -m "config-brancher --config-dir ci-operator/config --current-release=4.5 --future-release=4.6 --future-release=4.7 --confirm"

ci-operator-config-mirror --config-path ci-operator/config --to-org openshift-priv
git add ci-operator/config
git commit --signoff -S -m "ci-operator-config-mirror --config-path ci-operator/config --to-org openshift-priv"


cat <<EOF >/tmp/filter.py 
#!/usr/bin/env python3

import yaml
import sys
import copy

with open(sys.argv[1]) as f:
    all = yaml.full_load(f)
    for t in ("presubmits", "postsubmits"):
        for repo in all.get(t, {}):
            pruned = []
            for job in all.get(t, {}).get(repo, []):
                if job.get("agent", "") == "kubernetes":
                    pruned.append(job)
            all[t][repo] = pruned

with open(sys.argv[1], 'w') as f:
    yaml.dump(all, f, default_flow_style=False)
EOF

chmod +x /tmp/filter.py

for item in $(find ci-operator/jobs/ -name '*-release-4.6-*submits.yaml' ); do
    cp "${item}" "${item/4.6/4.7}"
    sed -i 's/-release-4.6-/-release-4.7-/g' "${item/4.6/4.7}" # job names
    sed -i 's/- release-4.6/- release-4.7/g' "${item/4.6/4.7}" # branches
    /tmp/filter.py "${item/4.6/4.7}"
done

ci-operator-prowgen --from-dir ci-operator/config --to-dir ci-operator/jobs
git add ci-operator/jobs
git commit --signoff -S -m "ci-operator-prowgen --from-dir ci-operator/config --to-dir ci-operator/jobs"

sanitize-prow-jobs --prow-jobs-dir ci-operator/jobs --config-path core-services/sanitize-prow-jobs/_config.yaml
git add ci-operator/jobs
git commit --signoff -S -m "sanitize-prow-jobs --prow-jobs-dir ci-operator/jobs --config-path core-services/sanitize-prow-jobs/_config.yaml"
```
/cc @stevekuznetsov 
/cc @openshift/openshift-team-developer-productivity-test-platform

### CODE REVIEW

File: ci-operator/jobs/kubernetes-reporting/metering-operator/kubernetes-reporting-metering-operator-release-4.7-postsubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+postsubmits:
+  kubernetes-reporting/metering-operator:
+  - agent: kubernetes
+    branches:
+    - ^master$
```
Comment (droslean) [inline ci-operator/jobs/kubernetes-reporting/metering-operator/kubernetes-reporting-metering-operator-release-4.7-postsubmits.yaml:5]: @stevekuznetsov Is this valid?

Comment (stevekuznetsov) [reply ci-operator/jobs/kubernetes-reporting/metering-operator/kubernetes-reporting-metering-operator-release-4.7-postsubmits.yaml]: No

**Comment by droslean** [comment]
rebased and regenerated after https://github.com/openshift/release/pull/8512 and https://github.com/openshift/release/pull/8515

File: ci-operator/jobs/kubevirt/hyperconverged-cluster-operator/kubevirt-hyperconverged-cluster-operator-release-4.7-postsubmits.yaml [hunk 2]
```diff
@@ -9,7 +9,6 @@ postsubmits:
       skip_cloning: true
     labels:
       ci-operator.openshift.io/prowgen-controlled: "true"
-    max_concurrency: 1
     name: branch-ci-kubevirt-hyperconverged-cluster-operator-release-4.7-images
```
Comment (petr-muller) [inline ci-operator/jobs/kubevirt/hyperconverged-cluster-operator/kubevirt-hyperconverged-cluster-operator-release-4.7-postsubmits.yaml:12]: Removing these lines looks suspicious.... What removed them? We added these for a reason.

**Comment by droslean** [comment]
I am using prowgen twice after generating also the private configs. Also I am running the `endsure_job_cluster.py` to update the cluster of the and after `sanitize-prow-jobs` to fix the jobs that `endsure_job_cluster.py` broke :dagger:

**Comment by droslean** [comment]
I will rerun the script and commit on each step to make the review easier.

**Comment by droslean** [comment]
I need to recreate the jobs after https://github.com/openshift/release/pull/8520

**Comment by droslean** [comment]
rebased and jobs have been recreated.

**Comment by droslean** [comment]
Notes: This PR should have included the changes that were done in https://github.com/openshift/release/pull/8547

**Pull request merged**